ultraplot/tests/test_subplots.py

```diff
@@ -172,3 +172,44 @@ def test_reference_aspect():
     fig.auto_layout()
     assert np.isclose(refwidth, axs[fig._refnum - 1]._get_size_inches()[0])
     return fig
+
+
+@pytest.mark.mpl_image_compare
+@pytest.mark.parametrize("share", ["limits", "labels"])
+def test_axis_sharing(share):
+    fig, ax = uplt.subplots(ncols=2, nrows=2, share=share, span=False)
+    labels = ["A", "B", "C", "D"]
+    for idx, axi in enumerate(ax):
+        axi.scatter(idx, idx)
+        axi.set_xlabel(labels[idx])
+        axi.set_ylabel(labels[idx])
+
+    # TODO: the labels are handled in a funky way. The plot looks fine but the label are not "shared" that is the labels still exist but they are not visible and instead there are new labels created. Need to figure this out.
```

Comment:
```suggestion
    # TODO: the labels are handled in a funky way. The plot looks fine but
    # the label are not "shared" that is the labels still exist but they
    # are not visible and instead there are new labels created. Need to
    # figure this out.
```

Reply:
Do we want multiple lines or just rely on the editor(s)?

Reply:
multiple lines